src/quickfix.c

```diff
@@ -3261,8 +3261,35 @@ qf_jump_edit_buffer(
 		prev_winid == curwin->w_id ? curwin : NULL);
     }
     else
+    {
+	if (!forceit && curwin->w_p_wfb)
+	{
+	    if (qi->qfl_type == QFLT_LOCATION)
+	    {
+	        // Location lists cannot split or reassign their window
+	        // so 'winfixbuf' windows must fail
+	        semsg("%s", e_winfixbuf_cannot_go_to_buffer);
+	        return QF_ABORT;
+	    }
+
+	    if (win_valid(prevwin))
+	        // Change the current window to another because 'winfixbuf' is enabled
+	        curwin = prevwin;
+	    else
+	    {
+	        // Split the window, which will be 'nowinfixbuf', and set curwin to that
+	        exarg_T new_eap;
+	        CLEAR_FIELD(new_eap);
+	        new_eap.cmdidx = CMD_split;
+	        new_eap.cmd = (char_u *)"split";
+	        new_eap.arg = (char_u *)"";
+	        ex_splitview(&new_eap);
```

Comment:
Should already be used as a part of `Test_cnext_make_new_window`. We call `s:get_windows_count()`, then call `cnext` and confirm that a new window (`let l:expected = l:windows + 1`) was created. And then confirm that `cnext!` does not create a new window because we reuse the previously created split to subsequent `cnext` calls

Reply:
okay thanks